### PR TITLE
trick: use array destructuring to swap variables

### PR DESCRIPTION
**Pull request by wonderingabout**
credit for teaching me this trick :) https://www.typescriptlang.org/docs/handbook/variable-declarations.html#array-destructuring

### CODE REVIEW

File: 1-js/05-data-types/10-destructuring-assignment/article.md [hunk 1]
```diff
@@ -121,6 +121,44 @@ for (let [key, value] of user) {
...
+
+```js run
+let temp = a;
+a = b;
+b = temp;
+
+alert(`${a}, ${b}`); // 2, 1: successfully swapped!
+```
+
+But such code is long, unefficient, and error-prone. With deconstructing, we can do it in a much simpler way:
```
Comment (Dorus) [inline 1-js/05-data-types/10-destructuring-assignment/article.md:147]: unefficient -> inefficient

Comment (wonderingabout) [reply 1-js/05-data-types/10-destructuring-assignment/article.md]: fixed, ty :1st_place_medal:

**Comment by iliakan** [comment]
Sorry, that trick does not deserve such a long explanation.

I'd say, swapping variables is a very rare thing in practice. Maybe make it small?

**Comment by wonderingabout** [comment]
> Sorry, that trick does not deserve such a long explanation.
> 
> I'd say, swapping variables is a very rare thing in practice. We could add it though, if it were 3+ times smaller =)

sure, i'll submit a much smaller with a 1 commit rebasing long explanation, then a 1 commit short 
 explanation (so that we can compare short vs long PR)

**Comment by wonderingabout** [comment]
what do you think of the new version @iliakan ?

**Comment by iliakan** [comment]
:rocket:

**Pull request merged**